app/services/cache/mapping/declaration/DeclarationBuilder.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.cache.mapping.declaration
+
+import forms.declaration.RepresentativeDetails
+import services.cache.ExportsCacheModel
+import services.mapping.declaration._
+import services.mapping.goodsshipment.GoodsShipmentBuilder
+import wco.datamodel.wco.dec_dms._2.Declaration
+import wco.datamodel.wco.declaration_ds.dms._2.{DeclarationFunctionalReferenceIDType, DeclarationTypeCodeType}
+
+object DeclarationBuilder {
+
+  val defaultFunctionCode = "9"
+
+  def build(exportsCacheModel: ExportsCacheModel): Declaration = {
+    val declaration = new Declaration()
+
+    declaration.setFunctionCode(FunctionCodeBuilder.build(defaultFunctionCode))
+    declaration.setFunctionalReferenceID(createDeclarationFunctionalReferenceIDType(exportsCacheModel))
+
+    val typeCode: Option[DeclarationTypeCodeType] = for {
```

Comment:
Just style/preference comment. Can be ignored :)

I'm thinking about using map here. for yield for one thing is probably more complicated

Reply:
I'll haver a look and try and refactor this based on your comments.